File: Homework/hw2/1_hybrid/hybrid_image_starter.py
import matplotlib.pyplot as plt
from align_image_code import align_images

# First load images

# high sf
im1 = plt.imread('./malik.png')/255.
im1 = im1[:,:,:3]

# low sf
im2 = plt.imread('./papadimitriou.png')/255
im2 = im2[:,:,:3]

# Next align images (this code is provided, but may be improved)
im1_aligned, im2_aligned = align_images(im1, im2)

## You will provide the code below. Sigma1 and sigma2 are arbitrary 
## cutoff values for the high and low frequencies

sigma1 = arbitrary_value_1
sigma2 = arbitrary_value_2
hybrid = hybrid_image(im1_aligned, im2_aligned, sigma1, sigma2)

plt.imshow(hybrid)
plt.show()

File: Homework/hw2/1_hybrid/hybrid_image_starter.m
%% Starter Code
close all; % closes all figures
% read images and convert to single format
im1 = im2single(imread('./malik.png'));
im2 = im2single(imread('./papadimitriou.png'));
im1 = rgb2gray(im1); % convert to grayscale
im2 = rgb2gray(im2);

% use this if you want to align the two images (e.g., by the eyes) and crop
% them to be of same size
% [im2, im1] = align_images(im2, im1);

% uncomment this when debugging hybridImage so that you don't have to keep aligning
% keyboard; 

%% Choose the cutoff frequencies and compute the hybrid image (you supply this code)
arbitrary_value = 100;
cutoff_low = 6;
cutoff_high = 16; 
im12 = hybrid_image(im1, im2, cutoff_low, cutoff_high);

%% Crop resulting image (optional)
figure(1), hold off, imagesc(im12), axis image, colormap gray
disp('input crop points');
[x, y] = ginput(2);  x = round(x); y = round(y);
im12 = im12(min(y):max(y), min(x):max(x), :);
figure(1), hold off, imagesc(im12), axis image, colormap gray